### PR TITLE
Add download utility specialized for torchaudio

### PR DESCRIPTION
**Pull request by mthrok**
In recent updates, torchaudio added features that download assets/models from
download.pytorch.org/torchaudio.

To reduce the code duplication, the implementations uses utilities from
``torch.hub``, but still, there are patterns repeated in implementing
the fetch mechanism, notably cache and local file path handling.

This commit introduces the utility function that handles
download/cache/local path management that can be used for 
fetching pre-trained model data.

### CODE REVIEW

File: torchaudio/utils/download.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        Currently the valid key values are the route on ``download.pytorch.org/torchaudio``,
+        but this is an implementation detail.
+
+    Returns:
+        str: The path to the asset on the local file system.
+    """
+    path = path or _get_local_path(key)
+
+    if path.exists():
+        _LG.info("The local file (%s) exists. Skipping the download.", path)
```
Comment (hwangjeff) [inline torchaudio/utils/download.py:72]: is there an advantage to doing this versus using f-strings?

**Review by hwangjeff — APPROVED** [review]
lg

**Review by carolineechen — APPROVED** [review]
looks good, thanks for adding this

File: torchaudio/utils/download.py (hunk 1 above)
Comment (mthrok) [reply torchaudio/utils/download.py:72]: The official advantage is that if a logging is not triggered, (say, the config level is set to WARNING by client code here so that info is not triggered), then the string substitution is not performed, so that we can avoid constructing the log message string.

https://docs.python.org/3/howto/logging.html#optimization

Personally I think it is negligible, but pylint with default configuration complaints this

https://pylint.pycqa.org/en/latest/messages/warning/logging-fstring-interpolation.html
